src/sentry/api/serializers/models/dashboard.py

```diff
@@ -63,18 +64,31 @@ class DashboardListSerializer(Serializer):
     def get_attrs(self, item_list, user):
         item_dict = {i.id: i for i in item_list}
 
-        widgets = list(
+        widgets = (
             DashboardWidget.objects.filter(dashboard_id__in=item_dict.keys())
             .order_by("order")
-            .values_list("dashboard_id", "order", "display_type")
+            .values("dashboard_id", "order", "display_type", "detail", "id")
         )
 
-        result = defaultdict(lambda: {"widget_display": []})
-        for dashboard_id, _, display_type in widgets:
-            dashboard = item_dict[dashboard_id]
-            display_type = DashboardWidgetDisplayTypes.get_type_name(display_type)
+        result = defaultdict(lambda: {"widget_display": [], "widget_preview": []})
+        for widget in widgets:
+            dashboard = item_dict[widget["dashboard_id"]]
+            display_type = DashboardWidgetDisplayTypes.get_type_name(widget["display_type"])
             result[dashboard]["widget_display"].append(display_type)
 
+        for widget in widgets:
```

Comment:
I chose two loops because it easily separates the logic between adding to `widget_display` and adding to `widget_preview`. I don't mind merging them if people feel strongly about it, but I felt like the performance penalty is low and this makes it easier to remove `widget_preview` down the line